flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java

```diff
@@ -946,16 +947,23 @@ private Operation convertShowViews(SqlShowViews sqlShowViews) {
     private Operation convertRichExplain(SqlRichExplain sqlExplain) {
         Operation operation;
         SqlNode sqlNode = sqlExplain.getStatement();
+        Set<String> explainDetails = sqlExplain.getExplainDetails();
+
+        //  Link to FLINK-22155,EXPLAIN statement should validate insert and query.If sql is a
+        //  INSERT statement, it will parse to RichSqlInsert. If sql is a SELECT statement
+        //  it will be converted to SqlSelect, but when this SELECT statement contains UNION
+        //  it will be converted to SqlBasicCall and it's operator is union instead of converted
+        //  to SqlSelect SqlNode.
         if (sqlNode instanceof RichSqlInsert) {
             operation = convertSqlInsert((RichSqlInsert) sqlNode);
-        } else if (sqlNode instanceof SqlSelect) {
+        } else if (sqlNode instanceof SqlSelect || sqlNode instanceof SqlBasicCall) {
```

Comment:
replace the condition with `sqlNode.getKind().belongsTo(SqlKind.QUERY)` as line 300.